<?xml version="1.0" ?>
<launch>
  <arg name="camera_name"	default="a_bot_camera"/>
  <arg name="camera_width"	default="640"/>
  <arg name="camera_height"	default="480"/>
  <arg name="camera_frame"	default="$(arg camera_name)_color_optical_frame"/>
  <arg name="plcf_dir"		default="$(find aist_localization)/plcf"/>
  <arg name="urdf_dir"
       default="$(find o2as_parts_description)/urdf/generated"/>

  <rosparam command="load"
	    file="$(find aist_localization)/config/models.yaml"/>

  <include file="$(find realsense2_camera)/launch/rs_camera.launch">
    <arg name="camera"			value="$(arg camera_name)"/>
    <arg name="enable_depth"		value="true"/>
    <arg name="enable_infra1"		value="false"/>
    <arg name="enable_infra2"		value="false"/>
    <arg name="enable_pointcloud"	value="false"/>
    <arg name="align_depth"		value="true"/>
    <arg name="depth_width"		value="$(arg camera_width)"/>
    <arg name="depth_height"		value="$(arg camera_height)"/>
    <arg name="infra_width"		value="$(arg camera_width)"/>
    <arg name="infra_height"		value="$(arg camera_height)"/>
    <arg name="color_width"		value="$(arg camera_width)"/>
    <arg name="color_height"		value="$(arg camera_height)"/>
  </include>

  <group ns="$(arg camera_name)">
    <include file="$(find aist_depth_filter)/launch/run.launch">
      <arg name="camera_info_topic"	value="color/camera_info"/>
      <arg name="image_topic"		value="color/image_raw"/>
      <arg name="depth_topic"		value="aligned_depth_to_color/image_raw"/>
      <arg name="normal_topic"		value=""/>
    </include>

    <node name="localization" pkg="aist_localization" type="aist_localization"
  	  output="screen">
      <remap from="/file_info"		to="depth_filter/file_info"/>
      <param name="camera_frame"	value="$(arg camera_frame)"/>
      <param name="plcf_dir"		value="$(arg plcf_dir)"/>
    </node>
  </group>

  <include file="$(find aist_model_spawner)/launch/run.launch">
    <arg name="urdf_dir"		value="$(arg urdf_dir)"/>
  </include>

  <arg name="command_args"
       value="-d $(find aist_localization)/launch/realsense.rviz"/>
  <node name="$(anon rviz)" pkg="rviz" type="rviz"
	respawn="false" args="$(arg command_args)"
	output="screen">
  </node>

  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>
</launch>
